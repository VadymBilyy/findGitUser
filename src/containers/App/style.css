.App {
  text-align: center;
}

.App-intro {
  font-size: large;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.top {
  background-color: yellow;
  width: 100%;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.bottom {
  display: flex;
  justify-content: center;
  flex-direction: column;
}

.usernameInput {
  border-radius: 4px;
  outline: none;
  border: 1px solid gray;
  margin-right: 10px;
  font-size: 20px;
  font-weight: bold;
}

.inputButton {
  flex: 0 0 auto;
  border-radius: 4px;
  width: 40px;
}

.inputContainer {
  display: flex;
  max-height: 30px;
  transform: translateY(-50%);
  justify-content: center;
}