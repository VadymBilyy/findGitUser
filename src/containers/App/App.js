import React, {Component} from 'react'
import { connect } from 'react-redux'
import {bindAll} from 'lodash'
import PropTypes from 'prop-types'
import findUserRequest from '../Requests/findUser'
import Profile from '../Profile/Profile'
import './style.css'

export class App extends Component {
    constructor(props) {
        super(props)

        bindAll(this, [
            'changeInputHandler' ,
            'checkUserHandler',
            'renderProfile',
            'onFetchingSuccess',
            'onFetchingFailure'
        ])

        this.state = {
            enteredName: ''
        }
    }


    changeInputHandler(e) {
        this.setState({
            enteredName: e.target.value
        })
    }

    checkUserHandler() {
        if (this.state.enteredName !== '') {
            this.props.dispatch({type: 'FETCHING_USER'})
            findUserRequest(this.state.enteredName)
                .then(res => this.onFetchingSuccess(res))
                .catch(err => this.onFetchingFailure(err))
        }
    }

    onFetchingSuccess(response) {
        const user = response.data
        this.props.dispatch({type: 'USER_FETCHED_SUCCESSFUL', data: {user}})
    }

    onFetchingFailure(error) {
        this.props.dispatch({type: 'USER_FETCHED_FAILURE', data: {error}})
    }

    renderProfile() {
        return (
            <div>
                {this.props.isProcessing && <h3>Loading...</h3>}
                {this.props.gitUser && !this.props.isProcessing ? <Profile gitUser={this.props.gitUser} /> : null}
            </div>
        )
    }


    render() {
        return (
          <div className="container">
            <div className="top">
              <h1>GitHub users search</h1>
            </div>
            <div className="bottom">
              <div className="inputContainer">
                <input type="text" className="usernameInput" onChange={this.changeInputHandler}/>
                <button className="searchButton" onClick={this.checkUserHandler}>Find</button>
              </div>
                {this.renderProfile()}
            </div>
          </div>
        )
    }
}


App.propTypes = {
    isProcessing: PropTypes.bool,
    dispatch: PropTypes.func,
    gitUser: PropTypes.shape({
        avatar_url: PropTypes.string,
        user: PropTypes.string,
        location: PropTypes.string,
        email: PropTypes.string,
        blog: PropTypes.string,
    })
}

export default connect(
        store => ({
            gitUser: store.profile.gitUser,
            isProcessing: store.profile.isProcessing
        }), dispatch => ({ dispatch })
)(App)
