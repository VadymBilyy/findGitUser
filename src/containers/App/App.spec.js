import React from 'react'
import { configure, mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import toJson from 'enzyme-to-json'
import {App} from './App'

configure({ adapter: new Adapter() })

describe('App tests', () => {
  test('Input user and start request', () => {
      const customDispatch = jest.fn()
      const component = mount(<App dispatch={customDispatch}/>)
      component.instance().onUserSuccess = jest.fn()
      component.instance().changeInputHandler({target: {value: 'boom'}})

      expect(component.instance().state.enteredName).toEqual('boom')
      component.instance().checkUserHandler()

      expect(customDispatch).toHaveBeenCalledWith({type: 'FETCHING_USER'})
  })

    test('Render component with isProcessing prop', () => {
        const component = shallow(<App isProcessing/>)
        expect(toJson(component)).toMatchSnapshot()
    })

    test('Render component with profile', () => {
        const customGitUser = {
            avatar_url: 'avatar_url',
            user: 'user',
            location: 'location',
            email: 'email',
            blog: 'blog'
        }
        const component = mount(<App isProcessing={false} gitUser={customGitUser} />)
        expect(toJson(component)).toMatchSnapshot()
    })
})