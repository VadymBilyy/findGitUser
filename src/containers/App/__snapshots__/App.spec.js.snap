// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Render component with isProcessing prop 1`] = `
<div
  className="container"
>
  <div
    className="top"
  >
    <h1>
      GitHub users search
    </h1>
  </div>
  <div
    className="bottom"
  >
    <div
      className="inputContainer"
    >
      <input
        className="usernameInput"
        onChange={[Function]}
        type="text"
      />
      <button
        className="searchButton"
        onClick={[Function]}
      >
        Find
      </button>
    </div>
    <div>
      <h3>
        Loading...
      </h3>
    </div>
  </div>
</div>
`;

exports[`App tests Render component with profile 1`] = `
<App
  gitUser={
    Object {
      "avatar_url": "avatar_url",
      "blog": "blog",
      "email": "email",
      "location": "location",
      "user": "user",
    }
  }
  isProcessing={false}
>
  <div
    className="container"
  >
    <div
      className="top"
    >
      <h1>
        GitHub users search
      </h1>
    </div>
    <div
      className="bottom"
    >
      <div
        className="inputContainer"
      >
        <input
          className="usernameInput"
          onChange={[Function]}
          type="text"
        />
        <button
          className="searchButton"
          onClick={[Function]}
        >
          Find
        </button>
      </div>
      <div>
        <Profile
          gitUser={
            Object {
              "avatar_url": "avatar_url",
              "blog": "blog",
              "email": "email",
              "location": "location",
              "user": "user",
            }
          }
        >
          <div
            className="Profile"
          >
            <img
              alt=""
              src="avatar_url"
            />
            <h2 />
            <div>
              location
            </div>
            <a
              href="email"
            >
              email
            </a>
            <a />
          </div>
        </Profile>
      </div>
    </div>
  </div>
</App>
`;
